### ISSUE TITLE
Play, pause and shuffle are always enabled

### ISSUE THREAD

**Issue by fstanis** [google/horologist#687]
1. Open media-sample
2. Observe "Nothing playing" at the top
3. Try pressing play
4. Observe that the button is enabled, but shouldn't be

This probably shouldn't happen: there's nothing to play, so the button should be disabled. This is also a `PlayerUiState` issue, not a compose issue and I suspect that ExoPlayer is advertising `COMMAND_PLAY_PAUSE` and `COMMAND_SET_SHUFFLE_MODE` even if there's nothing to play.

We should look into additional logic here, possibly checking if the player has any media items available.

**Comment by yschimke** [comment]
@fstanis If we confirm it's ExoPlayer, we should ask about it. Raise a bug, before blindly fixing assuming it's our fault.

It could be a bug over MediaController, but not direct ExoPlayer.

**Comment by fstanis** [comment]
I discussed this with the ExoPlayer team yesterday. The situation is a bit more complex.

`COMMAND_PLAY_PAUSE` and `COMMAND_SET_SHUFFLE_MODE` refer to the ability to change the value of `playWhenReady` and `shuffleModeEnabled` respectively. This is practically always possible, so these commands are always available.

What we (presumably) care about in the UI is whether "play" will eventually result in playing anything. This is only true if `playbackState` is in `STATE_READY` or `STATE_BUFFERING`. However, it's only possible to reach this state after calling `prepare()` and we currently do that when play is pressed, meaning our current implementation doesn't allow for knowing whether playing something is possible until we try.

The correct approach seems to be:
- Call `prepare()` more eagerly, possibly any time media is added / removed from the `Player`
- Disable play/pause unless the player is in `STATE_BUFFERING` or `STATE_READY`

**Comment by fstanis** [comment]
After exploring this further, I think we need some clearer guidelines on when to call `prepare()`.

Most notably, the play button in media apps often means "retry" when an error happens. However, errors will kick the player into `STATE_IDLE`, which would disable the play button if we implement the solution from my previous comment, making it impossible to retry playback (which requires another `prepare()`).

Perhaps a better option is to check `player.currentTimeline.isEmpty` to determine is play should be enabled and keep the current `prepare()` + `play()` behavior of the play button.

**Comment by yschimke** [comment]
![image](https://user-images.githubusercontent.com/231923/215819535-a38ddeb4-5143-4752-a8a3-9e11ab111749.png)
